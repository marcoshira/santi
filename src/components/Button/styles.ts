import styled, { css } from 'styled-components';

export const Wrapper = styled.button`
  ${({ theme }) => css`
    padding: ${theme.spacings.xxsmall} ${theme.spacings.xsmall};
    background: inherit;
    color: ${theme.colors.teal};
    /* border: 0.5px solid ${theme.colors.teal}; */
    border: none;
    font-size: ${theme.font.sizes.ltmedium};
    font-weight: 500;
    margin-left: ${theme.spacings.xxsmall};
    margin-top: 5px;
    margin-bottom: 4px;
    position: relative;
    transition: all 300ms ease-in-out;

    &:disabled {
      background: ${theme.colors.mediumGrey};
      color: ${theme.colors.white};
      border: 2px solid ${theme.colors.mediumGrey};
      transition: none;

      &:hover {
        background: ${theme.colors.mediumGrey};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.mediumGrey};
        cursor: not-allowed;
      }

      &::after {
        content: none;
      }
    }
    &:hover {
      color: ${theme.colors.thirdColor};
      cursor: pointer;
    }

    &::after {
      content: '';
      position: absolute;
      bottom: 0.2rem;
      left: 50%;
      width: 0;
      height: 0.2rem;
      background: ${theme.colors.thirdColor};
      transition: all 200ms ease-in-out;
    }

    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`;
