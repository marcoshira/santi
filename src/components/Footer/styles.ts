import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.2rem solid ${theme.colors.mediumGrey};

    a {
      color: inherit;
      text-decoration: none;
    }
  `}
`;
