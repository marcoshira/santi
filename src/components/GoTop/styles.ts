import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    position: fixed;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
    opacity: 0.7;
    border-radius: 2rem;
    border: 3px solid ${theme.colors.primaryColor};
  `}
`;
