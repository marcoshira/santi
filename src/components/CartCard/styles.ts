import styled, { css } from 'styled-components';
import { Wrapper as Btn } from '../Button/styles';
import { Title } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: row;
    width: 100%;
    overflow-x: hidden;
    box-shadow: 3px 3px 15px ${theme.colors.teal};
    border-radius: 15px;
    color: ${theme.colors.navy};
    background: ${theme.colors.footerWhite};
    position: relative;
    padding: ${theme.spacings.small} 0;
    margin-bottom: ${theme.spacings.ltmedium};

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      min-height: 50rem;
      width: 85%;
      justify-content: center;
      align-items: center;
      margin: ${theme.spacings.medium} auto;
    }

    a {
      cursor: pointer;
      &:hover {
        img {
          opacity: 0.6;
        }
      }
    }

    img {
      border-radius: 10px;
      height: 20rem;
      transition: ${theme.transitions.faster};

      @media ${theme.media.lteMedium} {
        height: 15rem;
      }
    }

    ${Title} {
      margin: 0;
    }
  `}
`;

export const ContentWrapper = styled.div`
  ${({ theme }) => css`
    width: 55%;
    a {
      text-decoration: none;
      color: ${theme.colors.navy};
      transition: ${theme.transitions.faster};
      &:hover {
        color: ${theme.colors.thirdColor};
      }
    }

    @media ${theme.media.lteMedium} {
      width: 100%;
      padding: ${theme.spacings.medium};
    }
  `}
`;

export const ImageWrapper = styled.div`
  display: flex;
  justify-content: center;
  width: 45%;
`;

export const ButtonContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    flex-direction: row;
    margin-top: 2rem;
    margin-right: 2rem;

    ${Btn} {
      min-width: 48%;

      margin: 0rem;
      margin-right: 1rem;
      border: 2px solid ${theme.colors.teal};
      background: ${theme.colors.footerWhite};
      color: ${theme.colors.teal};

      &:hover {
        border: 2px solid ${theme.colors.thirdColor};
        background: ${theme.colors.footerWhite};
        color: ${theme.colors.thirdColor};
      }

      &::after {
        content: none;
      }

      :nth-of-type(2) {
        background: ${theme.colors.teal};
        color: ${theme.colors.footerWhite};

        &:hover {
          border: 2px solid ${theme.colors.teal};
          background: ${theme.colors.hoverTeal};
        }
      }
    }

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      margin-right: 0rem;

      ${Btn} {
        width: 100%;
        margin: ${theme.spacings.tiny} 0;
      }
    }
  `}
`;
