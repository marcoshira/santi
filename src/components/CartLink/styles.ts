import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    position: absolute;
    left: 40rem;
    top: 9rem;
    width: calc(100vw - 55rem);

    @media ${theme.media.lteMedium} {
      width: 100%;
      left: auto;
      top: 6rem;
      margin: 0 auto;

      > ${Title} {
        margin-left: ${theme.spacings.ltlarge};
      }
    }
  `}
`;
