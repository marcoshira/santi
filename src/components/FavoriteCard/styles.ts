import styled, { css } from 'styled-components';
import { Wrapper as Btn } from '../Button/styles';
import { Title } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: row;
    width: 100%;
    box-shadow: 3px 3px 15px ${theme.colors.teal};
    border-radius: 15px;
    color: ${theme.colors.navy};
    background: ${theme.colors.footerWhite};
    position: relative;
    padding: ${theme.spacings.small} 0;
    margin-bottom: ${theme.spacings.ltmedium};

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      min-height: 53rem;
      width: 85%;
      justify-content: center;
      align-items: center;
      margin: ${theme.spacings.medium} auto;
      padding-bottom: 0;
    }

    a {
      cursor: pointer;
      &:hover {
        img {
          opacity: 0.6;
        }
      }
    }

    img {
      border-radius: 10px;
      height: 20rem;
      transition: ${theme.transitions.faster};

      @media ${theme.media.lteMedium} {
        height: 15rem;
      }
    }

    ${Title} {
      margin: 0;
    }
  `}
`;

export const ContentWrapper = styled.div`
  ${({ theme }) => css`
    width: 55%;
    a {
      text-decoration: none;
      color: ${theme.colors.navy};
      transition: ${theme.transitions.faster};
      &:hover {
        color: ${theme.colors.thirdColor};
      }
    }
    svg {
      height: 3rem;
      position: absolute;
      top: 2rem;
      right: 2rem;
      color: ${theme.colors.teal};
      cursor: pointer;
      transition: ${theme.transitions.fast};

      &:hover {
        color: ${theme.colors.hoverTeal};
      }
    }

    @media ${theme.media.lteMedium} {
      width: 100%;
      padding: ${theme.spacings.medium};
    }
  `}
`;

export const ImageWrapper = styled.div`
  ${({ theme }) => css`
    width: 45%;
    display: flex;
    justify-content: center;

    img {
    }
  `}
`;

export const ButtonContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-around;
    flex-direction: row;
    margin-top: 3rem;
    ${Btn} {
      min-width: 47%;

      margin: 0rem;
      margin-right: 1rem;
      border: 2px solid ${theme.colors.teal};
      background: ${theme.colors.footerWhite};
      color: ${theme.colors.teal};

      &:hover {
        border: 2px solid ${theme.colors.thirdColor};
        background: ${theme.colors.footerWhite};
        color: ${theme.colors.thirdColor};
      }

      &::after {
        content: none;
      }

      :nth-of-type(2) {
        background: ${theme.colors.teal};
        color: ${theme.colors.footerWhite};

        &:hover {
          border: 2px solid ${theme.colors.teal};
          background: ${theme.colors.hoverTeal};
        }
      }
    }

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      margin-right: 0rem;

      ${Btn} {
        width: 100%;
        margin: ${theme.spacings.tiny} 0;
      }
    }
  `}
`;

export const InputWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    width: 50%;
    justify-content: left;
    flex-flow: row nowrap;
    font-size: 1.6rem;
    margin-top: 3rem;

    @media ${theme.media.lteMedium} {
      width: 100%;
      margin-top: 1rem;
    }

    label {
      margin: 8px ${theme.spacings.ltlarge} 0 0;
      color: ${theme.colors.darkGrey};

      @media ${theme.media.lteMedium} {
        margin: 7px ${theme.spacings.hero} 0 0;
      }
    }

    input {
      width: 40px;
      display: flex;
      text-align: center;
      align-items: center;
      justify-content: center;
      border: 1px solid ${theme.colors.grey};
      height: 35px;
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    padding: 0 10px;
    background: ${theme.colors.footerWhite};
    border: 1px solid ${theme.colors.grey};
  `}
`;
