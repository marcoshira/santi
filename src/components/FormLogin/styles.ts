import styled, { css } from 'styled-components';
import { Wrapper as Button } from '../Button/styles';

export const Wrapper = styled.form``;

export const ButtonWrapper = styled.div`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.medium};
    display: flex;
    flex-direction: row;
    justify-content: space-between;

    ${Button} {
      a {
        text-decoration: none;
        color: ${theme.colors.footerWhite};
      }
      width: 15rem;
      border: 2px solid ${theme.colors.teal};
      background: ${theme.colors.teal};
      color: ${theme.colors.footerWhite};
      border-radius: ${theme.spacings.tiny};
      margin-left: 0;
      &:hover {
        background: ${theme.colors.hoverTeal};
      }
      &::after {
        height: 0;
      }
      &:disabled {
        background: ${theme.colors.mediumGrey};
        color: ${theme.colors.footerWhite};
        border: 2px solid ${theme.colors.mediumGrey};

        &:hover {
          background: ${theme.colors.mediumGrey};
          color: ${theme.colors.white};
          border: 2px solid ${theme.colors.mediumGrey};
          cursor: not-allowed;
        }

        &::after {
          content: none;
        }
      }

      @media ${theme.media.lteSmall} {
        width: 100%;
        padding: ${theme.spacings.xsmall} ${theme.spacings.xsmall};
      }
    }
  `}
`;
