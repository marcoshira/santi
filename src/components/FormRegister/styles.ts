import styled, { css } from 'styled-components';
import { Wrapper as Button } from '../Button/styles';

export const Wrapper = styled.form`
  ${({ theme }) => css``}
`;

export const ButtonWrapper = styled.div`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.medium};
    display: flex;
    flex-direction: row;
    justify-content: space-between;

    ${Button} {
      a {
        text-decoration: none;
        color: ${theme.colors.footerWhite};
      }
      width: 15rem;
      border: 2px solid ${theme.colors.teal};
      background: ${theme.colors.teal};
      color: ${theme.colors.footerWhite};
      border-radius: ${theme.spacings.tiny};
      margin-left: 0;
      &:hover {
        background: ${theme.colors.hoverTeal};
      }
      &::after {
        height: 0;
      }
      &:disabled {
        background: ${theme.colors.mediumGrey};
        color: ${theme.colors.footerWhite};
        border: 2px solid ${theme.colors.mediumGrey};

        &:hover {
          background: ${theme.colors.mediumGrey};
          color: ${theme.colors.white};
          border: 2px solid ${theme.colors.mediumGrey};
          cursor: not-allowed;
        }

        &::after {
          content: none;
        }
      }
    }
  `}
`;

export const ErrorMessage = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.thirdColor};
    font-size: ${theme.font.sizes.normal};
    margin-top: ${theme.spacings.small};
  `}
`;
