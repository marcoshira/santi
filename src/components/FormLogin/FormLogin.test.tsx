import { fireEvent, screen, waitFor } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { FormLogin } from '.';

const func = jest.fn();

describe('<FormLogin />', () => {
  it('should render', () => {
    renderTheme(<FormLogin />);

    expect(screen.getByLabelText('Email')).toBeInTheDocument();
  });
  it('should call function', async () => {
    renderTheme(<FormLogin onLogin={func} />);
    fireEvent.click(screen.getByRole('button'));

    await waitFor(() => {
      expect(func).toHaveBeenCalled();
    });
  });
  it('should change input', () => {
    renderTheme(<FormLogin />);
    const input = screen.getByLabelText('Email');
    const input2 = screen.getByLabelText('Password');
    fireEvent.change(input, { target: { value: 'abc' } });
    fireEvent.change(input2, { target: { value: 'abd' } });

    expect(input).toHaveValue('abc');
    expect(input2).toHaveValue('abd');
  });
  it('should render with errorMessage', () => {
    const { container } = renderTheme(<FormLogin errorMessage="oi" />);
    expect(screen.getByText('oi')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
