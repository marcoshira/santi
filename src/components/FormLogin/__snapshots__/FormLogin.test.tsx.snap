// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormLogin /> should render with errorMessage 1`] = `
.c0 {
  width: 100%;
  margin-top: 3.2rem;
}

.c1 {
  position: relative;
}

.c1 > svg {
  position: absolute;
  top: 50%;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
  right: 1rem;
  width: 2.5rem;
  height: 2.5rem;
  color: rgb(141,141,144);
  z-index: 1;
  color: rgb(247,38,115);
}

.c1 *:focus ~ svg {
  color: #8c4bff;
  color: rgb(247,38,115);
}

.c1 *:disabled ~ svg {
  color: rgb(201,201,203);
}

.c2 {
  border: 1px solid rgb(201,201,203);
  width: 100%;
  height: 100%;
  font-size: 1.8rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  padding: 1.6rem 1rem;
  background: #FFFFFF;
  padding-right: 3.7rem;
  border-radius: 0.4rem;
  outline: none;
  border: 0.1rem solid rgb(247,38,115);
  box-shadow: 0 0 0.2rem 0 rgb(247,38,115);
}

.c2::-webkit-input-placeholder {
  visibility: hidden;
  opacity: 0;
}

.c2::-moz-placeholder {
  visibility: hidden;
  opacity: 0;
}

.c2:-ms-input-placeholder {
  visibility: hidden;
  opacity: 0;
}

.c2::placeholder {
  visibility: hidden;
  opacity: 0;
}

.c2:focus {
  border: 0.1rem solid #8c4bff;
  box-shadow: 0 0 0.2rem 0 #8c4bff;
  background: #FFFFFF;
}

.c2:focus + .c3,
.c2:not(:placeholder-shown) + .c3,
.c2:-webkit-autofill + .c3 {
  top: 0;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
  font-size: 1.2rem;
  color: #FFFFFF;
  background: #8c4bff;
}

.c2:disabled {
  background: rgb(241,241,243);
  color: rgb(161,161,164);
}

.c2:disabled + .c3 {
  background: rgb(161,161,164);
  color: rgb(221,221,223);
}

.c2:focus {
  border: 0.1rem solid rgb(247,38,115);
  box-shadow: 0 0 0.2rem 0 rgb(247,38,115);
}

.c2:focus + .c3,
.c2:not(:placeholder-shown) + .c3 {
  color: #FFFFFF;
  background: rgb(247,38,115);
}

.c4 {
  display: inline;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
  left: 1rem;
  font-size: 1.6rem;
  height: 1.6rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  background: #FFFFFF;
  padding: 0.5rem 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  line-height: 1;
  z-index: 1;
  color: rgb(141,141,144);
  border-radius: 0.4rem;
}

.c6 {
  color: rgb(247,38,115);
  margin: 0.8rem 0 3.2rem;
}

.c5 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c9 {
  padding: 0.8rem 1rem;
  background: inherit;
  color: #8c4bff;
  border: none;
  font-size: 2rem;
  font-weight: 500;
  margin-left: 0.8rem;
  margin-top: 5px;
  margin-bottom: 4px;
  position: relative;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c9:disabled {
  background: #DDDDDD;
  color: #FFFFFF;
  border: 2px solid #DDDDDD;
  -webkit-transition: none;
  transition: none;
}

.c9:disabled:hover {
  background: #DDDDDD;
  color: #FFFFFF;
  border: 2px solid #DDDDDD;
  cursor: not-allowed;
}

.c9:disabled::after {
  content: none;
}

.c9:hover {
  color: #ff4a4a;
  cursor: pointer;
}

.c9::after {
  content: '';
  position: absolute;
  bottom: 0.2rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #ff4a4a;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
}

.c9:hover::after {
  left: 25%;
  width: 50%;
}

.c7 {
  margin-top: 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c7 .c8 {
  width: 15rem;
  border: 2px solid #8c4bff;
  background: #8c4bff;
  color: #FFFFFF;
  border-radius: 0.4rem;
  margin-left: 0;
}

.c7 .c8 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #FFFFFF;
}

.c7 .c8:hover {
  background: #6d39c7;
}

.c7 .c8::after {
  height: 0;
}

.c7 .c8:disabled {
  background: #DDDDDD;
  color: #FFFFFF;
  border: 2px solid #DDDDDD;
}

.c7 .c8:disabled:hover {
  background: #DDDDDD;
  color: #FFFFFF;
  border: 2px solid #DDDDDD;
  cursor: not-allowed;
}

.c7 .c8:disabled::after {
  content: none;
}

@media (max-width:440px) {
  .c7 .c8 {
    width: 100%;
    padding: 1rem 1rem;
  }
}

<div>
  <form
    class=""
  >
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <input
          class="c2"
          id="user-identifier"
          name="user-identifier"
          placeholder="Email"
          type="email"
          value=""
        />
        <label
          class="c3 c4"
          for="user-identifier"
        >
          Email
        </label>
        <svg
          aria-hidden="true"
          class="c5"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6zm-2 0l-8 5-8-5h16zm0 12H4V8l8 5 8-5v10z"
          />
        </svg>
      </div>
      <p
        class="c6"
      >
         
      </p>
    </div>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <input
          class="c2"
          id="user-password"
          name="user-password"
          placeholder="Password"
          type="password"
          value=""
        />
        <label
          class="c3 c4"
          for="user-password"
        >
          Password
        </label>
        <svg
          aria-hidden="true"
          class="c5"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M2 17h20v2H2v-2zm1.15-4.05L4 11.47l.85 1.48 1.3-.75-.85-1.48H7v-1.5H5.3l.85-1.47L4.85 7 4 8.47 3.15 7l-1.3.75.85 1.47H1v1.5h1.7l-.85 1.48 1.3.75zm6.7-.75l1.3.75.85-1.48.85 1.48 1.3-.75-.85-1.48H15v-1.5h-1.7l.85-1.47-1.3-.75L12 8.47 11.15 7l-1.3.75.85 1.47H9v1.5h1.7l-.85 1.48zM23 9.22h-1.7l.85-1.47-1.3-.75L20 8.47 19.15 7l-1.3.75.85 1.47H17v1.5h1.7l-.85 1.48 1.3.75.85-1.48.85 1.48 1.3-.75-.85-1.48H23v-1.5z"
          />
        </svg>
      </div>
      <p
        class="c6"
      >
        oi
      </p>
    </div>
    <div
      class="c7"
    >
      <button
        class="c8 c9"
      >
        Log In
      </button>
    </div>
  </form>
</div>
`;
